### PR TITLE
explore: init-send-inspect explicit NEAR Wallet loop

### PR DESCRIPTION
**Pull request by chadoh**
_I recommend viewing the diff [with indentation changes hidden](https://github.com/near/rainbow-bridge-frontend/pull/46/files?diff=unified&w=1)_

This explores some potential interfaces we could add to near-api-js to make it easier to code an "init-send-inspect" loop when an app expects the "send" step to cause a redirect to NEAR Wallet.

1. init: initialize transaction locally, storing transaction hash in local storage (in the code here, this hash gets stored within the `transfer` object).
2. send: the app needs to set URL Params prior to sending the transaction, to set things up for the "inspect" step.
3. inspect: the app has figured out that it just sent a transaction based on URL Params, and can easily fetch the same info about this transaction that would have been returned by a near-api-js contract call that required no redirect.

Note that, as coded in this particular app, this code would work whether or not the contract call causes a redirect.

_Again, I recommend viewing the diff [with indentation changes hidden](https://github.com/near/rainbow-bridge-frontend/pull/46/files?diff=unified&w=1)_

### CODE REVIEW

**Comment by chadoh** [comment]
This was my first exploration for a way to deal with potential NEAR Wallet redirects, and I did not expect it to be my favorite. But for this particular codebase, it ended up feeling the best. To compare to alternatives, see #47 & #48

File: src/js/transfers/bridged-nep21-to-erc20/index.js [hunk 1]
```diff
@@ -17,53 +23,97 @@ export async function initiate (nep21Address, amount) {
     { changeMethods: ['withdraw'] }
   )
 
-  // TODO: can cause redirect to NEAR Wallet, rendering `tx` inaccessible!
-  const tx = await bridgeToken.withdraw(
+  const withdrawTxHash = initializeTransaction(bridgeToken.withdraw(
```
Comment (vgrichina) [inline src/js/transfers/bridged-nep21-to-erc20/index.js:26]: I think this approach more or less can be implemented as is with current near-api-js:
```js
const [withdrawTxHash, withdrawTx] = account.signTransaction(bridgeTokenId, [functionCall('withdraw', {
       amount: String(amount),
       recipient: window.ethUserAddress.replace('0x', '')
     },
     new BN('3' + '0'.repeat(14)) // 10x current default from near-api-js
)]);
```

then persist either hash or transaction hash if needed and send using provider.

I guess main problem is that a lot of retry logic, etc is inside of `account.signAndSendTransaction`. Probably makes sense to move everything downstream (except nonce retries).

**Comment by chadoh** [comment]
from @vgrichina:

> seems like easiest to implement without making big near-api-js changes, but the downside is you have to deal with transactions explicitly in [every] case

**Comment by chadoh** [comment]
going with approach in #48

**Pull request closed without merging**